### ISSUE TITLE
BUG: MixedLM wald tests broken

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#2100]
t_test, f_test and wald_test raise the same exception

```
>>> res.wald_test(np.eye(3,5))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "e:\josef\eclipsegworkspace\statsmodels-git\statsmodels-all-new2_py27\statsmodels\statsmodels\base\model.py", line 1367, in wald_test
    LC = DesignInfo(names).linear_constraint(r_matrix)
  File "build\bdist.win32\egg\patsy\design_info.py", line 87, in __init__
ValueError: bad term slices

>>> len(res.params)
5

>>> res.cov_params().shape
(5, 5)
```

based on example for #2099 without including #2097 

```
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

data = sm.datasets.get_rdataset('dietox', 'geepack').data
mod = smf.mixedlm('Weight ~ Time', data, groups = 'Pig')
modf = mod.fit()
print modf.summary()

#When I try to include a random slope, the model runs and gets fitted:

mod  = smf.mixedlm('Weight ~ Time', data, groups = 'Pig', re_formula = '~Time')
modf = mod.fit()

#but I get the following error:

modf.model.data.param_names += ['r']

print modf.summary()
```

**Comment by josef-pkt** [comment]
I need to check and close for 0.7

**Comment by josef-pkt** [comment]
Ok, it works, at least the code is doing what it is supposed to be doing.

see #2231 for interpretation and wald tests that include random effects parameters

**Comment by josef-pkt** [comment]
Skipper's fix was backported to 0.6.1 (not 0.7 milestone)
